source/m2mfirmware.cpp

```diff
@@ -430,5 +430,7 @@ bool M2MFirmware::check_value_range(FirmwareResource resource, int64_t value) co
 
 void M2MFirmware::set_zero_value(M2MResource *resource)
 {
-    resource->set_value((const uint8_t*)"0", 1);
+    if(resource) {
+        resource->set_value((const uint8_t*)"0", 1);
+    }
```

Comment:
Maybe. Better than hiding the NULL. Usually I'm happy enough with the segmentation fault. Release images should have asserts compiled out anyway, and when debugging segmentation faults are pretty obvious.

Anyway, the Coverity warning will have been that you're not checking for NULL here, but are somewhere else on the path - means a logical inconsistency about whether you think the pointer can be NULL or not.

This if indicates you think it can be. An assert would indicat you think it can't be.